discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// metaLabelPrefix is the meta prefix used for all meta labels.
+// in this discovery.
+const (
+	metaLabelPrefix = model.MetaLabelPrefix + "scaleway_"
+	separator       = ","
+)
+
+// role is the role of the target within the Scaleway Ecosystem.
+type role string
+
+// The valid options for role.
+const (
+	// Scaleway Elements Baremetal
+	// https://www.scaleway.com/en/bare-metal-servers/
+	roleBaremetal role = "baremetal"
+
+	// Scaleway Elements Instance
+	// https://www.scaleway.com/en/virtual-instances/
+	roleInstance role = "instance"
+)
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *role) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal((*string)(c)); err != nil {
+		return err
+	}
+	switch *c {
+	case roleInstance, roleBaremetal:
+		return nil
+	default:
+		return errors.Errorf("unknown role %q", *c)
+	}
+}
+
+// DefaultSDConfig is the default Scaleway Service Discovery configuration.
+var DefaultSDConfig = SDConfig{
+	Port:            80,
+	RefreshInterval: model.Duration(60 * time.Second),
+	Zone:            scw.ZoneFrPar1.String(),
+	APIURL:          "https://api.scaleway.com",
+}
+
+type SDConfig struct {
+	// Project: The Scaleway Project ID used to filter discovery on.
+	Project string `yaml:"project"`
+
+	// APIURL: URL of the Scaleway API to use (will default to https://api.scaleway.com).
+	APIURL string `yaml:"api_url"`
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple scaleway_sd_configs
+	Zone string `yaml:"zone"`
+	// AccessKey used to authenticate on Scaleway APIs.
+	AccessKey string `yaml:"access_key"`
+	// SecretKey used to authenticate on Scaleway APIs.
+	SecretKey config.Secret `yaml:"secret_key"`
+	// FilterName to filter on during the ListServers.
+	FilterName string `yaml:"name"`
```

Comment:
```suggestion
	FilterName string `yaml:"name,omitempty"
```

This applies to all the non-mandatory fields